backend/src/api/workspaces.js

```diff
@@ -0,0 +1,103 @@
+const express = require('express')
+const { Workspace } = require('../models')
+const { directorsOnly, errorWrap } = require('../middleware')
+const router = express.Router()
+
+// Get all workspaces
+// TODO: when authentication server is integrated, only show the workspace that
+// belong to the caller of this endpoint.
+router.get(
+  '/',
+  [directorsOnly],
+  errorWrap(async (req, res) => {
+    const workspaces = await Workspace.find()
+    res.json({
+      code: 200,
+      result: workspaces,
+      success: true
+    })
+  })
+)
+
+// Find workspace by name
+// TODO: when authentication server is integrated, only show the workspace if it
+// belongs to the caller of this endpoint.
+router.get(
+  '/:workspaceName',
+  [directorsOnly],
+  errorWrap(async (req, res) => {
+    const workspaceName = req.params.workspaceName
+
+    if (!workspaceName) {
+      return res.json({
+        code: 400,
+        message: 'malformed request',
+        success: false
```

Comment:
I don't think this ever hits because if someone tries to hit the endpoint "/workspace/" it'll direct you to the first endpoint